velox/exec/fuzzer/PrestoQueryRunnerTimestampWithTimeZoneTransform.cpp

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "velox/exec/fuzzer/PrestoQueryRunnerTimestampWithTimeZoneTransform.h"
+#include "velox/functions/prestosql/types/TimestampWithTimeZoneType.h"
+#include "velox/parse/Expressions.h"
+#include "velox/type/tz/TimeZoneMap.h"
+#include "velox/vector/FlatVector.h"
+
+namespace facebook::velox::exec::test {
+std::string TimestampWithTimeZoneTransform::format(
+    const int64_t timestampWithTimeZone) const {
+  const auto timestamp = unpackTimestampUtc(timestampWithTimeZone);
+  const auto timeZoneId = unpackZoneKeyId(timestampWithTimeZone);
+  auto* timezonePtr = tz::locateZone(tz::getTimeZoneName(timeZoneId));
+
+  const auto maxResultSize = jodaDateTime_->maxResultSize(timezonePtr);
+  std::string str;
+  str.resize(maxResultSize);
+  const auto resultSize =
+      jodaDateTime_->format(timestamp, timezonePtr, maxResultSize, str.data());
+  str.resize(resultSize);
+
+  return str;
+}
+
+// Convert a Vector of TimestampWithTimeZone to a RowVector of BIGINT and
+// VARCHAR (millisUtc and time_zone).
```

Comment:
I don't find the piece of code that create a Bigint vector. Could you help point which line of code does this?

Reply:
My bad, I rewrote the code to produce just a VARCHAR with the formatted datetime (my original approach that this describes could not capture timestamps with millisecond precision), let me fix the comment.